modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AttachmentObjectFieldBusinessType.java

```diff
@@ -137,10 +137,15 @@ public Object getValue(
 			return fileEntryId;
 		}
 
-		JSONObject jsonObject = jsonFactory.createJSONObject(
-			MapUtil.getString(values, objectField.getName()));
+		if (value instanceof Map) {
+			fileEntryId = MapUtil.getLong((Map<String, Object>)value, "id");
+		}
+		else {
+			JSONObject jsonObject = jsonFactory.createJSONObject(
+				MapUtil.getString(values, objectField.getName()));
 
-		fileEntryId = GetterUtil.getLong(jsonObject.get("id"));
+			fileEntryId = GetterUtil.getLong(jsonObject.get("id"));
```

Comment:
Just a question: 

@p-albuquerque what's the case this code is executed? Do you know?

Reply:
@gabrielwas  The code inside **else** statement id executed, when the `values.get(objectField.getName())` receive a FileEntry entity. It happens when the attachment value don't be included in json of the PATCH request body.  So the headless fill the current FileEntry (and then, parse it to a JSONObject).
In this ticket scenarios, the `values.get(objectField.getName())` receive the FileEntry structure as a LinkedHashMap, that type don't fit with JSONObject.